### PR TITLE
Support type in module record options.

### PR DESCRIPTION
**Pull request by tmpfs**
Prepare to support ESM (via static module record) as well as CJS.

The value for `type` should be either `cjs` or `esm`.

### CODE REVIEW

**Review by kumavis — APPROVED** [review]

**Comment by kumavis** [comment]
currently using the `js` type in other parts of lavamoat, like handling the `builtins` in nodejs.

**Comment by tmpfs** [comment]
> currently using the `js` type in other parts of lavamoat, like handling the `builtins` in nodejs.

Thanks, I wasn't sure whether it was being used or just a placeholder. I have updated to use `js` here by default and in the bundler code.

File: packages/lavapack/src/runtime.js [hunk 1]
```diff
@@ -6423,13 +6423,13 @@ function makePrepareRealmGlobalFromConfig ({ createFunctionWrapper }) {
...
     // verify + load in each module
-    for (const [moduleId, moduleDeps, initFn, { package: packageName }] of newModules) {
+    for (const [moduleId, moduleDeps, initFn, { package: packageName, type }] of newModules) {
       // verify that module is new
       if (moduleRegistry.has(moduleId)) {
         throw new Error(`LavaMoat - loadBundle encountered redundant module definition for id "${moduleId}"`)
       }
       // add the module
-      moduleRegistry.set(moduleId, { type: 'js', id: moduleId, deps: moduleDeps, source: `(${initFn})`, package: packageName })
+      moduleRegistry.set(moduleId, { type: type || 'cjs', id: moduleId, deps: moduleDeps, source: `(${initFn})`, package: packageName })
```
Comment (kumavis) [inline packages/lavapack/src/runtime.js]: still cjs here

Comment (tmpfs) [reply packages/lavapack/src/runtime.js]: Oops, forgot to rebuild the runtime, done now :+1:

**Review by kumavis — APPROVED** [review]

**Pull request merged**